lighthouse-service/event_handler/evaluate_sli_handler.go

```diff
@@ -125,24 +128,24 @@ func (eh *EvaluateSLIHandler) processGetSliFinishedEvent(ctx context.Context, sh
 	if e.Result == "fail" {
 		evalResult.EventData.Result = keptnv2.ResultFailed
 		evalResult.Message = fmt.Sprintf("no evaluation performed by lighthouse because SLI failed with message %s", e.Message)
-		return sendEvent(shkeptncontext, triggeredID, keptnv2.GetFinishedEventType(keptnv2.EvaluationTaskName), eh.KeptnHandler, &evalResult)
+		return sendEvent(shkeptncontext, triggeredID, keptnv2.GetFinishedEventType(keptnv2.EvaluationTaskName), commitID, eh.KeptnHandler, &evalResult)
 	}
 
 	// compare the results based on the evaluation strategy
-	sloConfig, err := eh.SLOFileRetriever.GetSLOs(e.Project, e.Stage, e.Service)
+	sloConfig, err := eh.SLOFileRetriever.GetSLOs(e.Project, e.Stage, e.Service, commitID)
 
 	if err != nil {
 		if err == ErrSLOFileNotFound {
 			evalResult.EventData.Result = keptnv2.ResultPass
 			evalResult.Message = fmt.Sprintf("no evaluation performed by lighthouse because no SLO file configured for project %s", e.Project)
-			return sendEvent(shkeptncontext, triggeredID, keptnv2.GetFinishedEventType(keptnv2.EvaluationTaskName), eh.KeptnHandler, &evalResult)
+			return sendEvent(shkeptncontext, triggeredID, keptnv2.GetFinishedEventType(keptnv2.EvaluationTaskName), commitID, eh.KeptnHandler, &evalResult)
 		}
 
-		return sendErroredFinishedEventWithMessage(shkeptncontext, triggeredID, err.Error(), "", eh.KeptnHandler, e)
+		return sendErroredFinishedEventWithMessage(shkeptncontext, triggeredID, commitID, err.Error(), "", eh.KeptnHandler, e)
 	}
 
 	// get the slo.yaml as a plain file to avoid confusion due to defaulted values (see https://github.com/keptn/keptn/issues/1495)
-	sloFileContent, err := eh.KeptnHandler.GetKeptnResource("slo.yaml")
+	sloFileContent, err := eh.KeptnHandler.GetKeptnResource("slo.yaml") //TODO can I remove this now that we do have commitid retireval?
```

Comment:
I think yes, but can we make the `SLOFileRetriever.GetSLOs()` function also return the plain file content, in addition to the SLO object? See https://github.com/keptn/keptn/issues/1495 - there problem here was that some values of the SLO have been defaulted after unmarshalling, which led to some confusion, so we decided to also include the slo.yaml content exactly as it was uploaded to the configuration service